src/main/java/hk/com/novare/hellogit/services/UserFilesService.java

```diff
@@ -80,6 +80,18 @@ public User deleteUserByName(String name) {
         return null;
     }
 
+    public User updateUserByName(String name, User newUser){
+        for (User u : users) {
+            if (u.getName().equals(name)) {
+                newUser.setName(name);
+                users.remove(u);
```

Comment:
Removing and adding users while in the middle of a loop is unsafe. Does this work?